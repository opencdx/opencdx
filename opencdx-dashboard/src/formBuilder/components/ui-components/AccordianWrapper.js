import React from 'react';
import PropTypes from 'prop-types';
import Accordion from '@mui/material/Accordion';
import AccordionSummary from '@mui/material/AccordionSummary';
import AccordionDetails from '@mui/material/AccordionDetails';
import ExpandMoreIcon from '@mui/icons-material/ExpandMore';
import Typography from '@mui/material/Typography';

const AccordianWrapper = React.forwardRef(({ title, children }, ref) => {
    return (
        <Accordion ref={ref}>
            <AccordionSummary
                sx={{
                    backgroundColor: 'lightgray'
                }}
                expandIcon={<ExpandMoreIcon />}
                aria-controls="panel1a-content"
                id="panel1a-header"
            >
                <Typography>{title}</Typography>
            </AccordionSummary>
            <AccordionDetails>{children}</AccordionDetails>
        </Accordion>
    );
});
AccordianWrapper.propTypes = {
    title: PropTypes.string,
    children: PropTypes.node
};
export { AccordianWrapper };
