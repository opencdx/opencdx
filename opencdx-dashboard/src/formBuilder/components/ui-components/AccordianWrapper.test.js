import React from 'react';
import renderer from 'react-test-renderer';
import {AccordianWrapper} from './AccordianWrapper';

it('renders correctly', () => {
    const tree = renderer.create(<AccordianWrapper />).toJSON();
    expect(tree).toMatchSnapshot();
});
it('renders correctly with title', () => {
    const tree = renderer.create(<AccordianWrapper title="Test Title" />).toJSON();
    expect(tree).toMatchSnapshot();
} );
it('renders correctly with children', () => {
    const tree = renderer.create(<AccordianWrapper><div>Test</div></AccordianWrapper>).toJSON();
    expect(tree).toMatchSnapshot();
} );
it('renders correctly with title and children', () => {
    const tree = renderer.create(<AccordianWrapper title="Test Title"><div>Test</div></AccordianWrapper>).toJSON();
    expect(tree).toMatchSnapshot();
} );