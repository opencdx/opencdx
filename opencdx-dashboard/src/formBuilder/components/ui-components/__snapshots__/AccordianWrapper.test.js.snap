// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiAccordion-root MuiAccordion-rounded MuiAccordion-gutters css-1c35hjw-MuiPaper-root-MuiAccordion-root"
>
  <div
    aria-controls="panel1a-content"
    aria-expanded={false}
    className="MuiButtonBase-root MuiAccordionSummary-root MuiAccordionSummary-gutters css-x6j0gl-MuiButtonBase-root-MuiAccordionSummary-root"
    id="panel1a-header"
    onBlur={[Function]}
    onClick={[Function]}
    onContextMenu={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    role="button"
    tabIndex={0}
  >
    <div
      className="MuiAccordionSummary-content MuiAccordionSummary-contentGutters css-o4b71y-MuiAccordionSummary-content"
    >
      <p
        className="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
      />
    </div>
    <div
      className="MuiAccordionSummary-expandIconWrapper css-yw020d-MuiAccordionSummary-expandIconWrapper"
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
        data-testid="ExpandMoreIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M16.59 8.59 12 13.17 7.41 8.59 6 10l6 6 6-6z"
        />
      </svg>
    </div>
  </div>
  <div
    className="MuiCollapse-root MuiCollapse-vertical MuiCollapse-hidden css-bz4dnt-MuiCollapse-root"
    style={
      Object {
        "minHeight": "0px",
      }
    }
  >
    <div
      className="MuiCollapse-wrapper MuiCollapse-vertical css-smkl36-MuiCollapse-wrapper"
    >
      <div
        className="MuiCollapse-wrapperInner MuiCollapse-vertical css-9l5vo-MuiCollapse-wrapperInner"
      >
        <div
          aria-labelledby="panel1a-header"
          className="MuiAccordion-region"
          id="panel1a-content"
          role="region"
        >
          <div
            className="MuiAccordionDetails-root css-15v22id-MuiAccordionDetails-root"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`renders correctly with children 1`] = `
<div
  className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiAccordion-root MuiAccordion-rounded MuiAccordion-gutters css-1c35hjw-MuiPaper-root-MuiAccordion-root"
>
  <div
    aria-controls="panel1a-content"
    aria-expanded={false}
    className="MuiButtonBase-root MuiAccordionSummary-root MuiAccordionSummary-gutters css-x6j0gl-MuiButtonBase-root-MuiAccordionSummary-root"
    id="panel1a-header"
    onBlur={[Function]}
    onClick={[Function]}
    onContextMenu={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    role="button"
    tabIndex={0}
  >
    <div
      className="MuiAccordionSummary-content MuiAccordionSummary-contentGutters css-o4b71y-MuiAccordionSummary-content"
    >
      <p
        className="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
      />
    </div>
    <div
      className="MuiAccordionSummary-expandIconWrapper css-yw020d-MuiAccordionSummary-expandIconWrapper"
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
        data-testid="ExpandMoreIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M16.59 8.59 12 13.17 7.41 8.59 6 10l6 6 6-6z"
        />
      </svg>
    </div>
  </div>
  <div
    className="MuiCollapse-root MuiCollapse-vertical MuiCollapse-hidden css-bz4dnt-MuiCollapse-root"
    style={
      Object {
        "minHeight": "0px",
      }
    }
  >
    <div
      className="MuiCollapse-wrapper MuiCollapse-vertical css-smkl36-MuiCollapse-wrapper"
    >
      <div
        className="MuiCollapse-wrapperInner MuiCollapse-vertical css-9l5vo-MuiCollapse-wrapperInner"
      >
        <div
          aria-labelledby="panel1a-header"
          className="MuiAccordion-region"
          id="panel1a-content"
          role="region"
        >
          <div
            className="MuiAccordionDetails-root css-15v22id-MuiAccordionDetails-root"
          >
            <div>
              Test
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`renders correctly with title 1`] = `
<div
  className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiAccordion-root MuiAccordion-rounded MuiAccordion-gutters css-1c35hjw-MuiPaper-root-MuiAccordion-root"
>
  <div
    aria-controls="panel1a-content"
    aria-expanded={false}
    className="MuiButtonBase-root MuiAccordionSummary-root MuiAccordionSummary-gutters css-x6j0gl-MuiButtonBase-root-MuiAccordionSummary-root"
    id="panel1a-header"
    onBlur={[Function]}
    onClick={[Function]}
    onContextMenu={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    role="button"
    tabIndex={0}
  >
    <div
      className="MuiAccordionSummary-content MuiAccordionSummary-contentGutters css-o4b71y-MuiAccordionSummary-content"
    >
      <p
        className="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
      >
        Test Title
      </p>
    </div>
    <div
      className="MuiAccordionSummary-expandIconWrapper css-yw020d-MuiAccordionSummary-expandIconWrapper"
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
        data-testid="ExpandMoreIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M16.59 8.59 12 13.17 7.41 8.59 6 10l6 6 6-6z"
        />
      </svg>
    </div>
  </div>
  <div
    className="MuiCollapse-root MuiCollapse-vertical MuiCollapse-hidden css-bz4dnt-MuiCollapse-root"
    style={
      Object {
        "minHeight": "0px",
      }
    }
  >
    <div
      className="MuiCollapse-wrapper MuiCollapse-vertical css-smkl36-MuiCollapse-wrapper"
    >
      <div
        className="MuiCollapse-wrapperInner MuiCollapse-vertical css-9l5vo-MuiCollapse-wrapperInner"
      >
        <div
          aria-labelledby="panel1a-header"
          className="MuiAccordion-region"
          id="panel1a-content"
          role="region"
        >
          <div
            className="MuiAccordionDetails-root css-15v22id-MuiAccordionDetails-root"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`renders correctly with title and children 1`] = `
<div
  className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiAccordion-root MuiAccordion-rounded MuiAccordion-gutters css-1c35hjw-MuiPaper-root-MuiAccordion-root"
>
  <div
    aria-controls="panel1a-content"
    aria-expanded={false}
    className="MuiButtonBase-root MuiAccordionSummary-root MuiAccordionSummary-gutters css-x6j0gl-MuiButtonBase-root-MuiAccordionSummary-root"
    id="panel1a-header"
    onBlur={[Function]}
    onClick={[Function]}
    onContextMenu={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    role="button"
    tabIndex={0}
  >
    <div
      className="MuiAccordionSummary-content MuiAccordionSummary-contentGutters css-o4b71y-MuiAccordionSummary-content"
    >
      <p
        className="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
      >
        Test Title
      </p>
    </div>
    <div
      className="MuiAccordionSummary-expandIconWrapper css-yw020d-MuiAccordionSummary-expandIconWrapper"
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
        data-testid="ExpandMoreIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M16.59 8.59 12 13.17 7.41 8.59 6 10l6 6 6-6z"
        />
      </svg>
    </div>
  </div>
  <div
    className="MuiCollapse-root MuiCollapse-vertical MuiCollapse-hidden css-bz4dnt-MuiCollapse-root"
    style={
      Object {
        "minHeight": "0px",
      }
    }
  >
    <div
      className="MuiCollapse-wrapper MuiCollapse-vertical css-smkl36-MuiCollapse-wrapper"
    >
      <div
        className="MuiCollapse-wrapperInner MuiCollapse-vertical css-9l5vo-MuiCollapse-wrapperInner"
      >
        <div
          aria-labelledby="panel1a-header"
          className="MuiAccordion-region"
          id="panel1a-content"
          role="region"
        >
          <div
            className="MuiAccordionDetails-root css-15v22id-MuiAccordionDetails-root"
          >
            <div>
              Test
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
